test/treegen.lua

```diff
@@ -0,0 +1,129 @@
+-- Working tree generator.
+--
+-- Generates a tree of Lua files using provided templates and
+-- filenames.
+--
+-- Basic usage:
+--
+-- | local t = require('luatest')
+-- | local treegen = require('test.treegen')
+-- |
+-- | local g = t.group()
+-- |
+-- | local SCRIPT_TEMPLATE = [[
+-- |     <...>
+-- | ]]
+-- |
+-- | g.before_all(function(g)
+-- |     treegen.before_all_default(g)
+-- |     treegen.add_template(g, '^.*$', SCRIPT_TEMPLATE)
+-- | end)
+-- |
+-- | g.after_all(function(g)
+-- |     treegen.after_all_default(g)
+-- | end)
+-- |
+-- | g.foobar_test = function(g)
+-- |     local dir = treegen.prepare_directory(g,
+-- |         {'foo/bar.lua', 'main.lua'})
+-- |     <..test case..>
+-- | end
+
+local fio = require('fio')
+local log = require('log')
+
+local treegen = {}
+
+local function find_template(g, script)
+    for _, template_def in ipairs(g.templates) do
+        if script:match(template_def.pattern) then
+            return template_def.template
+        end
+    end
+    error(("treegen: can't find a template for script %q"):format(script))
+end
+
+-- Generate a script that follows a template and write it at given
+-- script file path in given directory.
+local function write_script(g, dir, script)
+    local script_abspath = fio.pathjoin(dir, script)
+    local flags = {'O_CREAT', 'O_WRONLY', 'O_TRUNC'}
+    local mode = tonumber('644', 8)
+    local fh = fio.open(script_abspath, flags, mode)
+    local template = find_template(g, script)
+
+    log.info(('Writing a script: %s'):format(script_abspath))
+    fh:write(template:gsub('<(.-)>', {
+        script = script,
+    }))
+
+    fh:close()
+end
+
+
+function treegen.before_all_default(g)
```

Comment:
`before_all_default()`/`after_all_defalt()` is slightly misleading to me.
Maybe `before_all_init()`/`after_all_clean()` is better?
Feel free to ignore.

Reply:
Agreed to name it `init` and `clean`.

Reply:
Renamed.